tiledb/cloud/client.py

```diff
@@ -375,148 +372,154 @@ def find_organization_or_user_for_default_charges(user):
-            self.set_disable_retries()
-        else:
-            raise Exception(
-                f"unsupported retry mode {mode!r}. Valid options are default, forceful or disabled"
-            )
-
-        self.update_clients()
-
-    def set_threads(self, threads=os.cpu_count() * 2):
-        """
-        Update thread pool sizes for async functionality
-        :param threads:
-        :return:
-        """
-        self.pool_threads = threads
-        self.update_clients()
-
-    def __get_array_api(self):
-        return rest_api.ArrayApi(
-            rest_api.ApiClient(
-                configuration=config.config, pool_threads=self.pool_threads
-            )
-        )
-
-    def __get_user_api(self):
-        return rest_api.UserApi(
-            rest_api.ApiClient(
-                configuration=config.config, pool_threads=self.pool_threads
-            )
-        )
-
-    def __get_organization_api(self):
-        return rest_api.OrganizationApi(
-            rest_api.ApiClient(
-                configuration=config.config, pool_threads=self.pool_threads
-            )
-        )
-
-    def __get_udf_api(self):
-        return rest_api.UdfApi(
-            rest_api.ApiClient(
-                configuration=config.config, pool_threads=self.pool_threads
-            )
-        )
-
-    def __get_tasks_api(self):
-        return rest_api.TasksApi(
-            rest_api.ApiClient(
-                configuration=config.config, pool_threads=self.pool_threads
-            )
-        )
-
-    def __get_sql_api(self):
-        return rest_api.SqlApi(
-            rest_api.ApiClient(
-                configuration=config.config, pool_threads=self.pool_threads
-            )
-        )
-
-    def __get_notebook_api(self):
-        return rest_api.NotebookApi(
-            rest_api.ApiClient(
-                configuration=config.config, pool_threads=self.pool_threads
-            )
-        )
+        self.retry_mode(RetryMode.FORCEFUL)
+
+    def retry_mode(self, mode: RetryOrStr = RetryMode.DEFAULT):
+        """Sets how we should retry requests and updates API instances."""
+        mode = RetryMode.maybe_from(mode)
+        config.config.retries = _RETRY_CONFIGS[mode]
+        client = rest_api.ApiClient(config.config)
+
+        self.array_api = rest_api.ArrayApi(client)
+        self.notebook_api = rest_api.NotebookApi(client)
+        self.organization_api = rest_api.OrganizationApi(client)
+        self.sql_api = rest_api.SqlApi(client)
+        self.tasks_api = rest_api.TasksApi(client)
+        self.udf_api = rest_api.UdfApi(client)
+        self.user_api = rest_api.UserApi(client)
+
+    def set_threads(self, threads: Optional[int] = None):
+        """Updates the number of threads in the async thread pool."""
+        with self._pool_lock:
+            old_pool = getattr(self, "_thread_pool", None)
+            self._thread_pool = futures.ThreadPoolExecutor(threads)
+            if old_pool:
+                old_pool.shutdown()
```

Comment:
This function is async right? The docs say that it blocks until all futures are done, so presumably it yields, but that's not explicitly clear.

Reply:
Whoops! I should have included `wait=False` here, so that it just tells the old pool to shut down. Thank you!